src/libtriton/arch/x86/x86Semantics.cpp

```diff
@@ -9913,6 +9919,132 @@ namespace triton {
       }
 
 
+      void x86Semantics::pmulhw_s(triton::arch::Instruction& inst) {
+        auto& src1 = inst.operands[0];
+        auto& src2 = inst.operands[1];
+
+        /* Create symbolic operands */
+        auto op1 = this->symbolicEngine->getOperandAst(inst, src1);
+        auto op2 = this->symbolicEngine->getOperandAst(inst, src2);
+
+        /* Create the semantics */
+        if ((!op1->isSymbolized() && op1->evaluate() == 0) ||
```

Comment:
Handling case when the result is concretely zeroed. Same in `pmulld_s`, `pmullw_s` and `vpmullw_s`, `vpmulhw_s`.